ajax/ajax.md

```diff
@@ -38,52 +32,76 @@ Front-end code of interactive websites often needs to talk to backend servers to
+| Difference Point        | Full Load Page          | AJAX  |
+| ------------- |:-----------------:| :---------:|
+| Client requests to server   | waits for response  | can perform other task simultaneously  |
+| Server Response       | responds a corresponding HTML page  |  responds with data (preferably JSON ) |
+| Render |  a complete new page is rendered   | only that part of page changes which has requested |
+| Thread Blocked | true   | false |
+| Data Received by | the browser in form of HTML   | JavaScript (callback or Promise) |
 
-### What does AJAX do?
+For better understanding follow this [Link](https://github.com/ashishnagpal2498/AjaxVsFullLoadPage)
 
-Let's think about how web applications work. Generally, the steps go like this:
-- The end user types a URL into a browser
-- The browser sends a request to a server designated by the URL
-- The server does some processing, probably queries a database, then responds to the browser by sending back HTML and JavaScript
-- The browser renders that HTML and JavaScript into a webpage
 
+### What does AJAX do?
 Let's say this website that the end user is visiting is an e-commerce site. The end user is browsing a list of products, sees a product they want to buy, and clicks the "Buy" button. This action (clicking the "Buy" button) triggers the request/response steps listed above, this time for the purpose of putting the product in the shopping cart.
 
 Before AJAX, the request/response cycle could only happen with the web page *as a whole*. The end user, by clicking the "Buy" button, would cause the backend server to generate a completely new set of HTML and JavaScript, and the browser would reload and render it as an entirely new page.
 
 AJAX, however, enabled the browser to make requests and only re-render *parts* of a webpage. This was beneficial performance-wise, because smaller strings of HTML, rather than all the HTML for the entire page, was all that was needed to be sent by the server and rendered by the browser. This also made the end user's browsing experience smoother: interactions with the webpage resulted in changes on screen without being interrupted by a reload.
 
-### Threads of Execution
-
-When your JavaScript code runs in the browser, it does so in a *single thread of execution*. What this means is that it runs your code in sequence, one command at a time. It cannot, for example, take two different lines in your code and run them simultaneously.
+### Remember
+[Thread of Execution](/javascript/javascript-async-9.md)
 
-To understand why asynchrony is an important feature of AJAX, we can think about an asynchronous scenario. Let's return to the example above of the end user visiting an e-commerce site. Imagine that the end user clicks on the "Buy" button to put an item in their shopping cart. This code might look something like this:
 
+### Hands-On Practice 
+* HTML
```

Comment:
Be explicit, telling them to create a directory, and what to name the html, css, and JS files.  Also, do they really need a css file, or can they just add this one style inline? As this is the AJAX outline, theres no real need to focus on style here.